modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,584 @@
+	}
+
+	public void setInputValue(String inputValue) {
+		_inputValue = inputValue;
+	}
+
+	public void setLabels(List<LabelItem> labels) {
+		_labels = labels;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	public void setSelectable(Boolean selectable) {
+		_selectable = selectable;
+	}
+
+	public void setSelected(Boolean selected) {
+		_selected = selected;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	public void setSpritemap(String spritemap) {
+		_spritemap = spritemap;
+	}
+
+	public void setSubtitle(String subtitle) {
+		_subtitle = subtitle;
+	}
+
+	public void setUserCard(UserCard userCard) {
+		_userCard = userCard;
+	}
+
+	public void setUserColorClass(String userColorClass) {
+		_userColorClass = userColorClass;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_actionDropdownItems = null;
+		_disabled = null;
+		_groupName = null;
+		_href = null;
+		_icon = null;
+		_imageAlt = null;
+		_imageSrc = null;
+		_inputName = null;
+		_inputValue = null;
+		_labels = null;
+		_name = null;
+		_selectable = null;
+		_selected = null;
+		_spritemap = null;
+		_subtitle = null;
+		_userCard = null;
+		_userColorClass = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/UserCard";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("actions", getActionDropdownItems());
+		props.put("description", getSubtitle());
+		props.put("disabled", getDisabled());
+		props.put("href", getHref());
+		props.put("id", getId());
+		props.put("inputName", getInputName());
+		props.put("inputValue", getInputValue());
+		props.put("userImageAlt", getImageAlt());
+		props.put("userImageSrc", getImageSrc());
+		props.put("userSymbol", getIcon());
+		props.put("labels", getLabels());
+		props.put("name", getName());
+		props.put("selectable", getSelectable());
+		props.put("selected", getSelected());
+		props.put("spritemap", getSpritemap());
+		props.put("userDisplayType", getUserColorClass());
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		Boolean selectable = getSelectable();
+
+		if ((selectable == null) || !selectable) {
+			cssClasses.add("card");
+		}
+
+		cssClasses.add("card-type-asset");
+		cssClasses.add("form-check");
+		cssClasses.add("form-check-card");
+		cssClasses.add("form-check-top-left");
+		cssClasses.add("user-card");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<div class=\"card\">");
+		jspWriter.write("<div class=\"aspect-ratio card-item-first\">");
+
+		Boolean disabled = getDisabled();
+		Boolean selectable = getSelectable();
+
+		if ((selectable != null) && selectable) {
```

Comment:
I think we can set defaults to match defaults from Clay:
```
private Boolean _disabled = false;
private Boolean _selectable = false;
private Boolean _selected = false;
...
```
This would simplify a lot of checks like this one.

Reply:
I think we can't due to the model usage to set attributes:
```
public Boolean getSelectable() {
		if ((_selectable == null) && (_userCard != null)) {
			return _userCard.isSelectable();
		}

		return _selectable;
	}
```
If we set defaults we can't check this

Reply:
We can just remove `_selectable == null` from that check, simplifying it further.

Reply:
Nope, because if you pass
`<clay:user-card selectable="<%= false %>"` or `<clay:user-card selectable="<%= true %>"`
that have priority over `_userCard`

Reply:
And if you're talking about
```
if ((selectable != null) && selectable) {
``
you can't remove `selectable != null` because if it's `null` and you try to evaluate it as a `boolean` that crashes in runtime.

Reply:
Actually we can just drop null check, because `getSelectable()` will never return null. It's because `_userCard.isSelectable()` will never return null, power of [default methods](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/soy/BaseClayCard.java#L79-L81) :muscle: 

Well, somebody can still set `<clay:user-card selectable="<%= null %>"` but that person deservers tag to fail :smile:

Reply:
Nope, still not good, doesn't cover if nothing is set
```
public Boolean getSelectable() {
	if ((_selectable == null) && (_userCard != null)) {
		return _userCard.isSelectable();
	}

	return (_selectable == null) ? false : _selectable;
}
```
That should do it. I think :)

Reply:
That looks better, thx @markocikos I'll give it a go!

Reply:
Just to clarify, this would achieve two things:
1. We would match defaults of Clay. There may be bugs if we don't render something instead of using default.
1. We can get rid of plentiful  `null` checks, e.g. `if ((selectable != null) && selectable) {` -> `if (selectable) {`

Reply:
Spoiler alert: Wall of text incoming.

## TL;DR
I've been thinking about this and I'd say is not that simple.

## PRIORITY OF TAG ATTRIBUTES VS MODEL ATTRIBUTES

Attributes passed to the tag have priority over those defined in the model, so given this use case:

```
userCard.setSelectable(false);

<clay:user-card
    selectable="<%= true %>"
    userCard="<%= userCard %>"
/>
```
In the end `selectable` must be `true`.

With this in mind we can see two different scenarios:

### Attribute default value is `null`

In this case there's no problem at all. In `getSelectable()` method we check if `_selectable == null`, if true we check if `_userCard != null` and if true we return the value from the model `return _userCard.isSelectable()`. If none of this is tru we just return the value of `_selectable`.

### Attribute default value is not `null`

Here we have a problem 'cause we have no way to know if the current value of `_selectable` is the default one or it's been passed. Let's take this example:

`_selectable` default value is `true` and in the use of the Tag we're passing `_selectable="<%= true %>`. We won't be able to check if `_selectable` is `true` because that's the default value or because that's the one we're receiving.

### So... what?

The solution I managed to think about with my limited JAVA knowledge would be to create properties in the class like `_defaultSelectable` and implement this kind of logic:

```
public Boolean getSelectable() {
    if (_selectable == null) {
        if (_userCard != null) {
            return _userCard.isSelectable();
        }
        else {
            return _defaultSelectable;
        }
    }

    return _selectable;
}

private Boolean _defaultSelectable = true;
private Boolean _selectable;
```

I have no idea if we can accomplish the same using some JAVA patter or mechanism, so I just leave it here for you to give me your thoughts 😄 


## DEFAULTS CONSISTENCY

Right now we're not being consisten in the defaults in the Tags, it's related Model and Clay. I'm talking in general but for the example I'll use UserCardTag and UserCard Model:

- In Clay: There's no `selectable`, we get the same result using `onChange` property, which default value is `null`
- In Soy `UserCardTag`: `selectable` default value is `null`
- In `UserCard` Model: `selectable` default value is `true`

So we need to decide what to do for this new `UserCardTag`, here's some scenarios:

- If we follow the default value defined in the Model, which would be `selectable = true` we're changing the tag old behavior, which would lead to unexpected behaviors if we don't migrate the use cases properly
- If we change `UserCard` Model default value to fit with the rest that (`selectable = null`) that will lead to unexpected behaviors if we don't update all the use cases to fit the new behavior, and that could be quite a lot because that default is defined in `BaseClayCard`, which is being extended by some other models.
- If we keep the same inconsistency so in the `UserCard` Model we keep `selectable = true` and in the `UserCardTag` we keep `selectable = false` we'll be doing the same we've been doing until now, but as I stated before that's not consistent due you'll get different default results using the tag passing the model vs not passing it.

I don't have any solution here, just leaving my train of thought here 😅

Reply:
> I have no idea if we can accomplish the same using some JAVA patter or mechanism, so I just leave it here for you to give me your thoughts 😄

There's no need for a `_defaultSelectable`, `return true` in that branch makes it default.

Reply:
> DEFAULTS CONSISTENCY

Ideally, all should be in sync as they should reflect our most common use case. We can cut Clay some slack since it's hard to anticipate DXP's most common use case precisely, so we should focus on the DXP side.

The most common use case now is through `UserCard` so that should what guides us now and the tag should follow those defaults imho.

Reply:
> There's no need for a _defaultSelectable variable. A simple return true in that branch makes it default.

Yeah, wouldn't that be less obvious and lead to confusion? (just asking)

Reply:
> The most common use case now is through UserCard so that should what guides us now and the tag should follow those defaults imho.

Ok, I'll change the defaults in the tag then

Reply:
> Yeah, wouldn't that be less obvious and lead to confusion? (just asking)

🤷 

If you wanted to signal default values, you'd better use constants and name the variables accordingly (uppercase). You'd want to be consistent for all defaults... I think it has merits, but I don't see this being a big deal. Declare them if you want, but make sure it's obvious they're constant and default values in the variable name.

Reply:
If you don't see it a big deal I'll leave the default in the return